cmd/wigStats/wigStats.go

```diff
@@ -0,0 +1,77 @@
+// Command Group: "Wig Tools"
+//TODO make wigStats compatible with negative wig values
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/wig"
+	"log"
+	"math"
+)
+
+func wigStats(inFile string, noGapFile string, outFile string, missingDataValue float64) {
+	var records []bed.Bed = bed.Read(noGapFile)
+	var w []wig.Wig = wig.Read(inFile)
+	var wigValues []float64
+	var statValues, tmpValues []int
+	var err error
+	var i, j int
+	statValues = make([]int, 1000)
+	for i = range records {
+		wigValues = wig.ChromToSlice(w, records[i].Chrom)
+		for j = records[i].ChromStart; j < records[i].ChromEnd; j++ {
+			if wigValues[j] == missingDataValue {
+				continue
+			}
+			if int(wigValues[j]) > len(statValues) {
+				tmpValues = make([]int, int(wigValues[j])-len(statValues)+1)
+				statValues = append(statValues, tmpValues...) // concatenate two slices together
+			}
+			statValues[int(wigValues[j])]++
+		}
+	}
+	out := fileio.EasyCreate(outFile)
+	_, err = fmt.Fprintf(out, "coverage\tcount\n")
+	exception.PanicOnErr(err)
+	for i = range statValues {
+		_, err = fmt.Fprintf(out, "%d\t%d\n", i, statValues[i])
+		exception.PanicOnErr(err)
+	}
+	err = out.Close()
+	exception.PanicOnErr(err)
+
+}
+
+func usage() {
+	fmt.Print(
+		"wigStats - provide coverage statistics for WIG format visualization files.\n" +
```

Comment:
If the requirements of the program can be specified in the usage that would be great, e.g. wig values must be non-negative integers (because stats are calculated by calling index numbers)

Reply:
I would also add that this can only handle fixed step Wig files. Also, I'm unable to tell, but can this handle wig files with a Step and/or Span other than "1"? If not, worth saying that here. And I would throw an error somewhere in the code if it is unable to handle this. 

On that note, I don't know if the wig is read in so that we can ask for a value at any coordinate if the Step and/or Span != 1? I see the wig.go file reads in Step and Span !=1, but I think it just stores the string, not that it creates a []int where each chr coordinate has a value. I think you would have to do this math in this code. Otherwise, a non-1 step/span would ruin the logic used on line 27 I believe. Technically, wig.Read can read in a Step and/or Span != 1, so I would again throw an error in this code and add this limitation to the usage statement. bedValueWig.go also has this limitation.